src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -2,41 +2,64 @@ import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Course, CourseScoreCalculationService, CourseService } from 'app/entities/course';
 import { ActivatedRoute } from '@angular/router';
 import { Subscription } from 'rxjs/Subscription';
-import { LangChangeEvent, TranslateService } from '@ngx-translate/core';
+import { TranslateService } from '@ngx-translate/core';
 import { HttpResponse } from '@angular/common/http';
 import * as moment from 'moment';
 import { Exercise, ExerciseService } from 'app/entities/exercise';
+import { AccountService } from 'app/core';
+import { sum } from 'lodash';
+
+enum ExerciseFilter {
+    OVERDUE = 'OVERDUE',
+    NEEDS_WORK = 'NEEDS_WORK',
+}
+
+enum ExerciseSortingOrder {
+    DUE_DATE_ASC = 1,
```

Comment:
I think it is clearer to have to store them as `ASC` and `DESC` strings and not numbers

Reply:
True, but we use the numbers to calculate sorting order when sorting the exercises, i.e. get a negative or positive value of the comparison depending on the order.